pkg/vpcmodel/nodesExplainability.go

```diff
@@ -164,17 +164,17 @@ func (c *VPCConfig) GetNodesOfVsi(vsi string) ([]Node, error) {
 //  2. Calculate from N and the cidr block, disjoint IP blocks
 //  3. Return the nodes created from each block from 2 contained in the input cidr
 func (c *VPCConfig) getCidrExternalNodes(cidr string) (cidrNodes []Node, err error) {
-	cidrsIPBlock := common.NewIPBlockFromCidrOrAddress(cidr)
-	if cidrsIPBlock == nil { // string cidr does not represent a legal cidr
-		return nil, nil
+	cidrsIPBlock, err := common.NewIPBlockFromCidrOrAddress(cidr)
+	if err != nil { // string cidr does not represent a legal cidr
+		return nil, err
```

Comment:
@ShiriMoran  should this be changed? (avoid returning the error returned from `NewIPBlockFromCidrOrAddress`, and just return a `nil` object) ?

Reply:
yes, because we don't know whether the user meant to insert an `IPBlock/Cidr`. If this failed all we know is that what he inserted does not match any legal src/dst. For example, we will get there if instead of vsi1-ky, VSI1-KY was inserted (we decided not be oblivious to cases in this stage) and the error message from `NewIPBlockFromCidrOrAddress` will be misleading. We could use some string matching to "guess" what the user meant to do, but this is surely not worth the effort for the 1st release

Reply:
The linter had this issue 
```
pkg\vpcmodel\explainabilityInput.go:136:3: error is not nil (line 133) but it returns nil 
```
so I've made a few more changes (I think it made sense, found it also confusing to return `nil` when `err` was not `nil`), so now this function is called only after validating that the input represents a valid IP address or CIDR.